providers/implementations/kdfs/pbkdf2.c

```diff
@@ -283,9 +313,15 @@ static int kdf_pbkdf2_set_ctx_params(void *vctx, const OSSL_PARAM params[])
 #endif
     }
 
-    if ((p = OSSL_PARAM_locate_const(params, OSSL_KDF_PARAM_PASSWORD)) != NULL)
+    if ((p = OSSL_PARAM_locate_const(params, OSSL_KDF_PARAM_PASSWORD)) != NULL) {
+        if (ctx->lower_bound_checks != 0
+            && p->data_size < KDF_PBKDF2_MIN_PASSWORD_LEN) {
+            ERR_raise(ERR_LIB_PROV, PROV_R_INVALID_KEY_LENGTH);
```

Comment:
If you use PROV_R_PASSWORD_SIZE_TOO_SMALL also here, and do `make update` the new error reason entry will be created.

Reply:
Why you didn't use the PROV_R_PASSWORD_STRENGTH_TOO_WEAK reason code here as well?